### PR TITLE
Prevent room and staff introductions in battles

### PR DESCRIPTION
**Pull request by ImLycan**
Technically it is possible, but talking w HoeenHero and we decided that its really impractical. So probably to get rid of it.

Any other commands that seem unnecessary is battles?

### CODE REVIEW

File: chat-commands.js [hunk 1]
```diff
@@ -1087,6 +1087,7 @@ const commands = {
 
 	topic: 'roomintro',
 	roomintro: function (target, room, user, connection, cmd) {
+		if (room.battle) return this.errorReply("Battles do not support room intros.");
```
Comment (ImLycan) [inline chat-commands.js]: maybe change the text?

**Comment by ImLycan** [comment]
Actually, me and Hoeen were talking more, and it may have to do with the editbattle permission in config

**Comment by HoeenCoder** [comment]
Mostly that players can create roomintro/edit the rules link. I wouldn't trust a random user to not say put an inappropriate image in the battle's roomintro and post the link in a bunch of places.

Looking at permissions in config-example.js https://github.com/Zarel/Pokemon-Showdown/blob/master/config/config-example.js#L394 editroom says its only for modjoin/privacy in groupchats/battles, which is outdated tbh since its used by alot of room owner level commands like blacklist, roomintro, ect.

Basically we could do a few things:
1. Prevent the commands we don't want used in battles with a `if (room.battle) some error reply` like lycan has now.
2. Update permissions, make `editbattle` a permission for players maybe and only use that on modjoin/modchat, and make `editroom` for Room Owner level commands.
3. Something else, heck we could remove `editroom` from players and add a exception to the permission check for players in the commands we want them to use.

Lmk what you guys think.

**Comment by Zarel** [comment]
That's fair. We could escalate certain commands to `makeroom`, and then give ROs an in-room `makeroom`. It would be kind of bad for the room though.

I'm fine with hardcoding "no roomintros for players".

**Comment by ImLycan** [comment]
I was able to do some further research and i was able to a bit more stuff, I edit the room rules, i was able to change the room language and other room setting commands without using /roomsettings by using stuff like /capsfilter on or /slowchat 60

**Comment by ImLycan** [comment]
So, i had a bit of trouble, mainly in finding something to use and i didn't know how to hard code a command to a rank. However, I decided to use the permission `joinbattle` since players can have it and drivers up. But I am not sure if this is the right way to go since there was no description for that specific permission.

**Comment by Zarel** [comment]
There's no permission specific to players?

**Comment by Zarel** [comment]
If so, we could create a new one for it.

**Comment by ImLycan** [comment]
I'll make a new permission for players called `battleprivacy` then.

**Comment by ImLycan** [comment]
Well, updated, i think i might've messed something up

File: config/config-example.js [hunk 2]
```diff
@@ -51,35 +51,11 @@ exports.wsdeflate = {
...
-
-/*
-Main's SSL deploy script from Let's Encrypt looks like:
-	cp /etc/letsencrypt/live/sim.psim.us/privkey.pem ~user/Pokemon-Showdown/config/ssl/
-	cp /etc/letsencrypt/live/sim.psim.us/fullchain.pem ~user/Pokemon-Showdown/config/ssl/
-	chown user:user ~user/Pokemon-Showdown/config/ssl/privkey.pem
-	chown user:user ~user/Pokemon-Showdown/config/ssl/fullchain.pem
-*/
+ * TODO: allow SSL to actually be possible to use for third-party servers at
+ * some point.
```
Comment (Zarel) [inline config/config-example.js]: botched rebaseeeee

File: config/config-example.js [hunk 3]
```diff
@@ -51,35 +51,11 @@ exports.wsdeflate = {
...
 
 /**
- * proxyip - proxy IPs with trusted X-Forwarded-For headers
+  * proxyip - proxy IPs with trusted X-Forwarded-For headers
```
Comment (Zarel) [inline config/config-example.js]: here too

File: config/config-example.js (hunk 2 above)
Comment (ImLycan) [reply config/config-example.js]: Yea I kinda fucked up. But i was able to revert that commit thank god

Comment (ImLycan) [reply config/config-example.js]: Also im curious on how you were able to review that outdated commit?

Comment (Zarel) [reply config/config-example.js]: I clicked on it in notifications before you reverted it, I think.

**Comment by Zarel** [comment]
idk if I like `battleprivacy` as a name.

Maybe something like `editprivacy`

**Comment by HoeenCoder** [comment]
Also give ‘editprivacy’ to leaders so we can continue to edit modjoin/hidden room status in battles.

**Comment by Zarel** [comment]
Actually is there any reason for `editroom` and `editprivacy` to be different permissions?

**Comment by Zarel** [comment]
Oh, lol, nvm, that's how this got started in the first place.

**Comment by ImLycan** [comment]
So uh anything else you want me to do? because i was able to test and it works again

**Comment by ImLycan** [comment]
Sorry but, bump?
